catroid/src/test/java/org/catrobat/catroid/test/web/WebConnectionHolderTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.web;
+
+import org.catrobat.catroid.web.WebConnection;
+import org.catrobat.catroid.web.WebConnectionHolder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+
+@RunWith(JUnit4.class)
+public class WebConnectionHolderTest {
+
+	@Test
+	public void testAddConnectionToList() {
+		WebConnection webConnection = Mockito.mock(WebConnection.class);
+		WebConnectionHolder webConnectionHolder = new WebConnectionHolder();
+		boolean isConnectionSuccessfullyAddedToList = webConnectionHolder.addConnectionToList(webConnection);
+
+		assertTrue(isConnectionSuccessfullyAddedToList);
+		assertEquals(1, webConnectionHolder.getConnections().size());
+	}
+
+	@Test
+	public void testAddElevenConnectionsToList() {
+		WebConnection webConnection = Mockito.mock(WebConnection.class);
+		WebConnectionHolder webConnectionHolder = new WebConnectionHolder();
+
```

Comment:
please dont...
just create a new list from array with 10 mocks of webConnection and set it in the connection holder and then add one to see that the 11th returns false.

tests should have a single purpose and test case, which should be easily spottable when reading the test. this is not at all the case with this one.

Reply:
with the above comment i actually meant using Collections.nCopies, or Arrays.fill, or anything but 10 adds.

Reply:
Fixed.